eval/src/tests/eval/tensor_lambda/tensor_lambda_test.cpp

```diff
@@ -77,14 +78,14 @@ void verify_impl(const vespalib::string &expr, const vespalib::string &expect) {
     verify_impl<T>(expr, expect, [](const T*){});
 }
 
-void verify_generic(const vespalib::string &expr, const vespalib::string &expect,
-                    EvalMode expect_eval_mode)
-{
-    verify_impl<DenseLambdaFunction>(expr, expect,
-                                     [&](const DenseLambdaFunction *info)
-                                     {
-                                         EXPECT_EQUAL(info->eval_mode(), expect_eval_mode);
-                                     });
+void verify_not_optimized(const vespalib::string &expr, const vespalib::string &expect) {
+    EvalFixture fixture(prod_factory, expr, param_repo, true);
+    EvalFixture simple_factory_fixture(simple_factory, expr, param_repo, false);
+    EXPECT_EQUAL(fixture.result(), simple_factory_fixture.result());
+    EXPECT_EQUAL(fixture.result(), EvalFixture::ref(expr, param_repo));
+    EXPECT_EQUAL(fixture.result(), EvalFixture::ref(expect, param_repo));
```

Comment:
use TensorSpec::from_expr here

Reply:
some of the "expect" expressions reference the param_repo, so that didn't work.